routes/api/v1/playlistSongs.js

```diff
@@ -41,6 +41,31 @@ const createPlaylistSong = router.post("/:id/favorites/:favoriteId", (request, r
     })
 })
 
+const deletePlaylistSong = router.delete("/:id/favorites/:favoriteId", (request, response) => {
+  const playlistId = request.params.id;
+  const songId = request.params.favoriteId;
+
+  database("playlist_songs").where("song_id", songId)
+    .then(favorite => {
+      if (favorite[0]) {
+        database("playlists").where({id: playlistId})
+          .then(playlist => {
+            if (playlist[0]) {
+              database("playlist_songs").del().where({ song_id: songId, playlist_id: playlistId })
```

Comment:
I think we need one more sad path catch in here- checking to make sure that an existing song is in the existing playlist.

Reply:
Good catch! I've fixed them and will push up the updates shortly